### ISSUE TITLE
How does IOReadCx / IOWriteCx get invoked?

### ISSUE THREAD

**Issue by markjfisher** [AppleWin/AppleWin#1256]
I'm trying to get AppleWin to work with FujiNet, as per #1201 

My initial work is here: https://github.com/markjfisher/AppleWin/blob/fujinet/source/FujiNet.cpp#L91
It's based on some changes Eric Carr did to integrate AppleWin to support a subset of functionality for FujiNet, and piggy backs the Uthernet2 work.

Whereas Eric's work interacts solely with C0Bx (slot 3, C0 handler), my code needs to work at C3xx.
I'm having trouble invoke my CX00 handler for my slot 3 device.

The code to invoke C0B0 read/writes works fine, whenever 6502 code reads/writes to C0 slot 3 (ie. C0Bx) then the handler is called.

However for C3xx addresses, I see 2 calls initially to C307 and C305 on bootup:

```
FUJINET cXHandler PC: f8bf, address: c307, write: 0, value: 0, cycles: 40, slot: 3
FUJINET IORead: PC: f8bf, address: c307
FUJINET cXHandler PC: 2207, address: c305, write: 0, value: 0, cycles: 94, slot: 3
FUJINET IORead: PC: 2207, address: c305
```

but after this, when I run my 6502 code, only the C0 handler ever is called, and any of my reads to C3xx do not cause the handling code in AppleWin to run.

What I'm trying to do is provide the thinnest SmartPort emulation possible, so that any code that calls this device is given all the correct signals (c301=$20, c303=$00, c305=$03, c307=$00), and a dispatch address so that too can be intercepted and a connection made with an external application that does the heavy lifting of emulating the FujiNet.
I'm an absolute beginner in the Apple2 world, so my knowledge is very limited, so the above may not work.

Can I get some help understanding why this is, and how I can fix it? Or what I need to understand as to why this can/cannot work?

Many thanks.
Mark

**Comment by tomcw** [comment]
Hi Mark,

I'd need to see your 6502 code, but I guess that you are not setting the memory soft switches correctly to either:
- disable INTCXROM, or
- enable SLOT3ROM

EG. (from the comment at the top of Memory.cpp):
```
 $C006   W       INTCXROMOFF     Enable slot ROM from $C100-$C7FF (but $C800-$CFFF depends on INTC8ROM)
 $C007   W       INTCXROMON      Enable main ROM from $C100-$CFFF
  : 
 $C00A   W       SLOTC3ROMOFF    Enable main ROM from $C300-$C3FF
 $C00B   W       SLOTC3ROMON     Enable slot ROM from $C300-$C3FF
```

If you haven't configured the MMU to use slot 3 ROM, then you'll just read from the Apple //e's main (internal) ROM... ie. your `CXHandler()` won't get called.

Side note: In the `SS_CARDTYPE` enum ([here](https://github.com/markjfisher/AppleWin/blob/fujinet/source/Card.h)), can you put the `CT_FujiNet` at the end. I need to add a comment at the top of this enum saying not the change the values, and only add new enums at the end. (The reason being these get persisted to the Registry when defining the user's card config, so these values must not change in future.)

**Comment by markjfisher** [comment]
Thanks for this, this is very helpful, however I went a different route.
(The 6502 code wasn't doing anything other than getting values from memory locations to prove the C++ handler would be invoked for the appropriate addresses.)

After chatting with some more knowledgeable Apple2 folk in FujiNet Discord, I moved to using slot 5 via the command line instead of piggy backing the Uthernet2 slot 3 mechanism, and it's now invoking my code correctly.

However, now I have to work out how to appease the boot sequence gods, as it's crashing after reading the initial C505, c503 c501, c5ff, c507, c5fe sequence. I have to go away and work out what it needs me to reply with correctly.
